### PR TITLE
[SHAD-334] Emit app metrics

### PR DESCRIPTION
**Pull request by robertofrontado**
### Jira
https://vilua.atlassian.net/browse/SHAD-334

### Context
Emitting 2 events (Only when the user is authenticated):
- HOST_APP_OPEN
- RECCO_SDK_OPEN

I chose to use UINotificationCenter to be able to hook into the host application lifecycle without requiring the client to do any extra steps. The other options I considered were to expose some lifecycle methods and good ol' swizzling 😅  none were optimal for an SDK.

### Media
https://github.com/SignificoHealth/recco-ios-sdk/assets/11177510/684d1c0e-6e78-4dfb-8f03-d8d14fd46ebb

### CODE REVIEW

File: Sources/ReccoHeadless/Repo/Repositories/MetricRepo.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+final class LiveMetricRepository: MetricRepository {
+    private let keychain: KeychainProxy
+
+    init(keychain: KeychainProxy) {
+        self.keychain = keychain
+    }
+
+    func log(event: AppUserMetricEvent) {
+        guard keychain.read(account: KeychainKey.clientUserId.rawValue) != nil else { return }
```
Comment (robertofrontado) [inline Sources/ReccoHeadless/Repo/Repositories/MetricRepo.swift:15]: @arubio @carmelojcortes No estoy seguro si esto tendrá algún impacto en performance, me imagino que sería mejor cachear el keychain en memoria para evitar leer de ahí todo el rato?

File: Sources/ReccoHeadless/Repo/Repositories/MetricRepo.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ final class LiveMetricRepository: MetricRepository {
     }
 
     func log(event: AppUserMetricEvent) {
-        guard keychain.read(account: KeychainKey.clientUserId.rawValue) != nil else { return }
+        guard keychain.read(account: KeychainKey.currentUser.rawValue) != nil else { return }
```
Comment (robertofrontado) [inline Sources/ReccoHeadless/Repo/Repositories/MetricRepo.swift:15]: @arubio @carmelojcortes No estoy seguro si esto tendrá algún impacto en performance, me imagino que sería mejor cachear el keychain en memoria para evitar leer de ahí todo el rato?

Comment (arubio) [reply Sources/ReccoHeadless/Repo/Repositories/MetricRepo.swift:15]: Keychain calls are just a simple sql select statement, no? In this case as the device is unlocked, the Keychain db will be unencrypted, so I don't think there will be a negative performance impact. We can test it nonetheless, if you think we should.

**Review by arubio — APPROVED** [review]

File: Sources/ReccoHeadless/Repo/Repositories/MetricRepo.swift (hunk 2 above)
Comment (robertofrontado) [reply Sources/ReccoHeadless/Repo/Repositories/MetricRepo.swift:15]: Got it, that makes sense, I read somewhere that it was super fast for small amounts of data, which is what we store in this case, so probably its all good 👍 we can revisit it later if we end up having performance problems

**Pull request merged**